l10n_nl_tax_declaration_reporting/report/report_vat.py

```diff
@@ -0,0 +1,288 @@
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+from openerp.osv import fields
+from openerp import api, models
+
+
+class vat_report(models.AbstractModel):
+    _name = 'report.l10n_nl_tax_declaration_reporting.tax_report_template'
+
+    @api.multi
+    def render_html(self, data=None):
+        self.cr = self._cr
+        self.uid = self._uid
+        report_obj = self.env['report']
+        report = report_obj._get_report_from_name(
+            'l10n_nl_tax_declaration_reporting.tax_report_template')
+
+        res = {}
+        self.period_ids = []
+        period_obj = self.pool.get('account.period')
+        self.display_detail = data['form']['display_detail']
+        _chart_tax_id = data['form']['chart_tax_id']
+        _chart_tax_name = self.env['account.tax.code'].search(
+            [('id', '=', _chart_tax_id)]).name
+
+        _fiscalyear_id = data['form']['fiscalyear_id']
+        _fiscalyear_name = self.env['account.fiscalyear'].search(
+            [('id', '=', _fiscalyear_id)]).name
+
+        _period_from_id = data['form']['period_from']
+        if _period_from_id:
+            _period_from_name = self.env['account.period'].search(
+                [('id', '=', _period_from_id)]).name
+        else:
+            _period_from_name = 'Begin fy'
+
+        _period_to_id = data['form']['period_to']
+        if _period_to_id:
+            _period_to_name = self.env['account.period'].search(
+                [('id', '=', _period_to_id)]).name
+        else:
+            _period_to_name = 'End fy'
+
+        res['periods'] = ''
+        if data['form'].get('period_from', False) and\
+                data['form'].get('period_to', False):
+            self.period_ids = period_obj.build_ctx_periods(
+                self.cr, self.uid, data['form']['period_from'],
+                data['form']['period_to'])
+            periods_l = period_obj.read(
+                self.cr, self.uid, self.period_ids, ['name'])
+            for period in periods_l:
+                if res['periods'] == '':
+                    res['periods'] = period['name']
+                else:
+                    res['periods'] += ", " + period['name']
+
+        self._get_lines(data['form']['based_on'], data['form']['company_id'])
+        self._5AB = int(self._1AB + self._1BB + self._1CB +
```

Comment:
This should be no issue accoung the tax authourity http://www.belastingdienst.nl/wps/wcm/connect/bldcontentnl/belastingdienst/zakelijk/btw/administratie_bijhouden/facturen_maken/btw-bedrag_afronden
Here is stated that you may round subtotals and do it in your own advantage.